addons/crm/static/src/js/forecast/forecast_model_extension.js

```diff
@@ -0,0 +1,117 @@
+ *  today:          2021-04-21
+ *  granularity:    month
+ *  field name:     date_field
+ *  -> the domain would be: ['&', ['date_field', '=', false], ['date_field', '>=', '2021-04-01']]
+ */
+class ForecastModelExtension extends ActionModel.Extension {
+
+    //---------------------------------------------------------------------
+    // Public
+    //---------------------------------------------------------------------
+
+    /**
+     * @override
+     * @returns {any}
+     */
+    get(property) {
+        switch (property) {
+            case "domain": return this.getDomain();
+        }
+    }
+
+    /**
+     * @override
+     */
+    prepareState() {
+        Object.assign(this.state, {
+            forecastField: null,  // {string} forecast_field from the context (name)
+            forecastFieldType: null,  // {string} forecast_field_type (date or datetime)
+            forecastFilter: null,  // {boolean} if the "Forecast" filter is active
+            forecastGranularity: null,  // {string} the granularity on the forecast_field groupby (year, quarter, month, week, day, hour)
+        });
+    }
+
+    /**
+     * The state needs to be recomputed each time the parent model initiates a load action.
+     * @override
+     * @returns {Promise}
+     */
+    callLoad() {
+        this._computeState();
+        return super.callLoad(...arguments);
+    }
+    
+    //---------------------------------------------------------------------
+    // Actions / Getters
+    //---------------------------------------------------------------------
+
+    /**
+     * Adds a domain constraint to only get records from the start of the period containing "now",
+     * only if the "Forecast" filter is active
+     * @returns {Array[]}
+     */
+    getDomain() {
+        if (!this.state.forecastFilter) {
+            return null;
+        }
+        var startMoment = moment().startOf(this.state.forecastGranularity);
+        var format = this.state.forecastFieldType === 'datetime' ? "YYYY-MM-DD HH:mm:ss" : "YYYY-MM-DD";
+        return ['|', [this.state.forecastField, '=', false], [this.state.forecastField, '>=', startMoment.format(format)]];
+    }
+
+    //---------------------------------------------------------------------
+    // Private
+    //---------------------------------------------------------------------
+
+    /**
+     * The state is mostly dependent on 2 context keys :
+     * forecast_field -> is a prerequisite for this extension to work
+     * forecast_filter -> a filter which applies this context key should be present and active in order to get the
+     * modified domain
+     * If the forecast_field is present in the groupby, the related granularity is used
+     *   If no granularity is specified, the default is "month"
+     * Else, "day" is used (the filter will get data from 00:00:00 today)
+     */
+    _computeState() {
+        this.prepareState();
+        if (typeof this.config.context.forecast_field === 'undefined') {
```

Comment:
Maybe `if (! this.config.context.forecast_field) { return; }` ? Do we really need to compare to undefined ?